### PR TITLE
Publishing wasm modules into org registry

### PR DESCRIPTION
**Pull request by rumatoest**
This will build and publish wasm modules wrapped in docker container on each commit to main  or tag (that is not -test or -dev).

### CODE REVIEW

File: .github/workflows/build-test-publish.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - "!**-test"
+      - "!**-dev"
+
+
+jobs:
+  build-modules:
+    uses: infinity-swap/.github/.github/workflows/build-n-test.yml@v0.1
+    with:
+      test-script: ./scripts/cargo.test.sh
+      skip-test: ${{ github.ref_type == 'tag' }}
```
Comment (blutooth) [inline .github/workflows/build-test-publish.yml:23]: I don't think we want to skip tests.

File: .github/workflows/build-test-publish.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      output-artifact: is20-modules
+    secrets:
+      gh_token: ${{ secrets.GH_PKG_TOKEN }}
+      gh_login: ${{ secrets.GH_PKG_LOGIN }}
+  
+  publish-prepare:
```
Comment (blutooth) [inline .github/workflows/build-test-publish.yml:29]: I think the logic for publich-prepare can be moved to the .github publishing workflow as the checks such as 
 ${{ github.ref_type == 'branch' && github.ref == 'refs/heads/main' }} will be possible.

File: .github/workflows/build-test-publish.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      
+
+  publishing:
+    needs: [build-modules, publish-prepare]
+    uses: infinity-swap/.github/.github/workflows/publish-wrapped-artifact.yml@v0.1
+    with:
+      input-artifact: is20-modules
+      output-image-name: is20-modules
+      output-image-tag: ${{ needs.publish-prepare.outputs.tag }}
+      output-image-tag-latest: ${{ needs.publish-prepare.outputs.latest == 'true' }}
```
Comment (blutooth) [inline .github/workflows/build-test-publish.yml:53]: Can remove these vars : 
output-image-tag: ${{ needs.publish-prepare.outputs.tag }}
output-image-tag-latest: ${{ needs.publish-prepare.outputs.latest == 'true' }}

File: .github/workflows/build-test-publish.yml (hunk 2 above)
Comment (rumatoest) [reply .github/workflows/build-test-publish.yml:30]: Probably but I have no idea how to pass output variables from reusable forflow to the next job

Comment (rumatoest) [reply .github/workflows/build-test-publish.yml:30]: Found how to do it.
Yes it is possible to hide some logic but still we should pass all input arguments there

**Comment by rumatoest** [comment]
I have no idea how to produce dependency-like cargo artifact with all wasms and did files. So I need this to be published for CI pipelines

**Pull request closed without merging**